tsinfer/inference.py

```diff
@@ -584,6 +549,228 @@ def match_ancestors(
     return ts
 
 
+def match_ancestors_batch_init(
+    working_dir,
+    sample_data_path,
+    ancestor_data_path,
+    min_work_per_job,
+    *,
+    sgkit_samples_mask_name=None,
+    sites_mask_name=None,
+    recombination_rate=None,
+    mismatch_ratio=None,
+    path_compression=True,
+    # Deliberately undocumented parameters below
+    recombination=None,  # See :class:`Matcher`
+    mismatch=None,  # See :class:`Matcher`
+    precision=None,
+    engine=constants.C_ENGINE,
+    extended_checks=False,
+    time_units=None,
+    record_provenance=True,
+):
+    # Create work dir
+    pathlib.Path(working_dir).mkdir(parents=True, exist_ok=True)
+    ancestors = formats.AncestorData.load(ancestor_data_path)
+    sample_data = formats.SgkitSampleData(
+        sample_data_path,
+        sgkit_samples_mask_name=sgkit_samples_mask_name,
+        sites_mask_name=sites_mask_name,
+    )
+    ancestors._check_finalised()
+    sample_data._check_finalised()
+
+    matcher = AncestorMatcher(
+        sample_data,
+        ancestors,
+    )
+    ancestor_grouping = []
+    ancestor_lengths = ancestors.ancestors_length
+    for group_index, group_ancestors in matcher.group_by_linesweep().items():
+        # Make ancestor_ids JSON serialisable
+        group_ancestors = list(map(int, group_ancestors))
+        partitions = []
+        current_partition = []
+        current_partition_work = 0
+        # TODO: Can do better here by packing ancestors
+        # into as equal sized partitions as possible
+        if group_index == 0:
+            partitions.append(group_ancestors)
+        else:
+            total_work = sum(ancestor_lengths[ancestor] for ancestor in group_ancestors)
+            min_work_per_job_group = min_work_per_job
+            if total_work / 1000 > min_work_per_job:
```

Comment:
Probably good to cover this somehow, it's the sort of thing that would catch us out in real applications. Could make the ``1000`` a parameter, and then test with something small?